dataprofiler/profilers/profile_builder.py

```diff
@@ -1374,22 +1400,45 @@ def _update_profile_from_chunk(self, data, sample_size,
         elif isinstance(data, list):
             data = pd.DataFrame(data)
 
-        if len(data.columns) != len(data.columns.unique()):
-            raise ValueError('`StructuredProfiler` does not currently support '
-                             'data which contains columns with duplicate '
-                             'names.')
+        duplicate_cols_given = len(data.columns) != len(data.columns.unique())
+
+        # Error out if schema given does not match schema present in data
+        if self._initialized:
+            mapping_given = dict()
+            for i in range(len(data.columns)):
+                col = data.columns[i]
+                if isinstance(col, str):
```

Comment:
maybe mention a comment above this suggest that pandas by default will give DFs an integer value for the name if columns aren't specified hence why this is needed